crates/sui-indexer/src/handlers/objects_snapshot_processor.rs

```diff
@@ -62,105 +92,220 @@ impl Default for SnapshotLagConfig {
     }
 }
 
-// NOTE: "handler"
-impl<S> ObjectsSnapshotProcessor<S>
+#[async_trait]
+impl<S, T> Worker for ObjectsSnapshotProcessor<S, T>
```

Comment:
I see, indexer's CheckpointHandler will process the entire checkpoint, meanwhile we just need the ObjectChanges.

Out of scope for this PR, but down the line might be interesting to modify CheckpointHandler so someone can customize their indexer to process some combination of data from each checkpoint - so rather than implement `Worker` trait, they can just use `CheckpointHandler`